src/fuzz_introspector/analysis.py

```diff
@@ -544,49 +544,58 @@ def detect_branch_level_blockers(
+                if int(line_number) > int(side_line_number):
+                    logger.debug(
+                        "Branch-blocker: Anomalous branch sides line nubmers: %s:%s -> %s" % (
+                            source_file_path, line_number, side_line_number
+                        )
+                    )
                     continue
 
-            hitcount_diff = max(sides_hitcount + [branch_hitcount])
-            link = fuzz_profile.resolve_coverage_link(
-                target_coverage_url,
-                source_file_path,
-                int(line_number),
-                function_name
-            )
-            new_blk = FuzzBranchBlocker(blocked_side, blocked_unique_not_covered_com,
-                                        blocked_unique_reachable_com, blocked_unique_funcs,
-                                        blocked_not_covered_com, blocked_reachable_com,
-                                        hitcount_diff, source_file_path, line_number,
-                                        side_line_number, function_name, link)
-            fuzz_blockers.append(new_blk)
+                # Sanity check for fall through cases: checks if the branch side has coverage or not
+                if coverage.get_type() == "file":
+                    if coverage.is_file_lineno_hit(source_file_path, int(side_line_number)):
+                        logger.debug("Branch-blocker: fall through branch side is not blocked: %s"
+                                     % (side_line))
+                        continue
+                else:
+                    if coverage.is_func_lineno_hit(function_name, int(side_line_number)):
+                        logger.debug("Branch-blocker: fall through branch side is not blocked: %s"
+                                     % (side_line))
+                        continue
+
+                hitcount_diff = max(sides_hitcount + [branch_hitcount])
+                link = fuzz_profile.resolve_coverage_link(
+                    target_coverage_url,
+                    source_file_path,
+                    int(line_number),
+                    function_name
+                )
+                new_blk = FuzzBranchBlocker(blocked_side, blocked_unique_not_covered_com,
+                                            blocked_unique_reachable_com, blocked_unique_funcs,
+                                            blocked_not_covered_com, blocked_reachable_com,
+                                            hitcount_diff, source_file_path, line_number,
+                                            side_line_number, function_name, link)
+                fuzz_blockers.append(new_blk)
+            except Exception:
+                # Temporary fix for https://github.com/google/oss-fuzz/issues/9209
+                # This should be refined and made more precise.
+                logger.debug("Branch-blocker: exception occurred in iteration of not-taken-sides")
```

Comment:
Let's log the `llvm_branch.sides` as well to help finding the cause.

Reply:
Done, please check

Reply:
Thanks!